pallet/src/lib.rs

```diff
@@ -49,20 +91,75 @@ pub mod pallet {
 
     #[pallet::genesis_build]
     impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
-        fn build(&self) {
-            Pallet::<T>::initialize_authorities(&self.authorities);
-        }
+        fn build(&self) {}
     }
-}
 
-impl<T: Config> Pallet<T> {
-    fn initialize_authorities(authorities: &[T::AuthorityId]) {
-        if !authorities.is_empty() {
-            assert!(
-                <Authorities<T>>::get().is_empty(),
-                "Authorities are already initialized!"
-            );
+    impl<T: Config> Pallet<T> {
+        fn initialize_authorities(authorities: &[T::AuthorityId]) {
+            if !authorities.is_empty() {
+                assert!(
+                    <Authorities<T>>::get().is_empty(),
+                    "Authorities are already initialized!"
+                );
+                <Authorities<T>>::put(authorities);
+            }
+
+            <SessionInfo<T>>::put(SessionChange {
+                session_id: 0,
+                changed: true,
```

Comment:
If you set `changed` to `true` here, doesn't it result in setting the authorities again in `on_finalize`?

Reply:
Yes but only once for the first block and it will only update the store state to the same state it had. Other options I saw was some ugly boolean flags with if-else which I thought are not needed for such a one-time case.